patches/third_party-WebKit-Source-modules-battery-NavigatorBattery.cpp.patch

```diff
@@ -0,0 +1,16 @@
+diff --git a/third_party/WebKit/Source/modules/battery/NavigatorBattery.cpp b/third_party/WebKit/Source/modules/battery/NavigatorBattery.cpp
+index 2083421bbe21327f8c1252437387bbe9926abc56..ab702ca8ce9f36d0aa1daad09b6b6dc6ee2fe5d0 100644
+--- a/third_party/WebKit/Source/modules/battery/NavigatorBattery.cpp
++++ b/third_party/WebKit/Source/modules/battery/NavigatorBattery.cpp
+@@ -16,7 +16,11 @@ NavigatorBattery::NavigatorBattery(Navigator& navigator)
+ 
+ ScriptPromise NavigatorBattery::getBattery(ScriptState* script_state,
+                                            Navigator& navigator) {
++#if defined(MUON_CHROMIUM_BUILD)
+   return NavigatorBattery::From(navigator).getBattery(script_state);
++#else 
++  return ScriptPromise{};
```

Comment:
I don't think just an empty script promise will work correctly. In browser-laptop we call `reject`

Reply:
for this I'm wondering if we should override `BatteryMonitorImpl::DidChange` to always return 100%. @diracdeltas ?

Reply:
@bridiver for webcompat, it's usually better if we pretend the API isn't supported at all (vs returning a fake value)

Reply:
well, to accomplish that in browser laptop without breaking things we called reject in promise so I think we need to do `return ScriptPromise::Reject(script_state, v8::Exception::TypeError(...))`

Reply:
I also think it's probably better to replace the idl file if we're going to do things like this and have our own class for `ImplementedAs`